pkg/clusterapi/controlplane.go

```diff
@@ -13,7 +13,7 @@ import (
 )
 
 // ControlPlane represents the provider-specific spec for a CAPI control plane using the kubeadm CP provider
-type ControlPlane[C, M Object] struct {
+type ControlPlane[C Object[C], M Object[M]] struct {
```

Comment:
Checking my understanding here: `Object` itself is now a generic, and we pass `[C]` and `[M]` respectively to specify Object's type?

Reply:
addressed offline